share/p2p/shrexnd/client.go

```diff
@@ -141,22 +141,15 @@ func (c *Client) doRequest(
 func convertToNamespacedShares(rows []*pb.Row) (share.NamespacedShares, error) {
 	shares := make([]share.NamespacedRow, 0, len(rows))
 	for _, row := range rows {
-		var proof *ipld.Proof
+		var proof *nmt.Proof
 		if row.Proof != nil {
-			cids := make([]cid.Cid, 0, len(row.Proof.Nodes))
-			for _, node := range row.Proof.Nodes {
-				cid, err := cid.Cast(node)
-				if err != nil {
-					return nil, fmt.Errorf("casting proofs node to cid: %w", err)
-				}
-				cids = append(cids, cid)
-			}
-
-			proof = &ipld.Proof{
-				Nodes: cids,
-				Start: int(row.Proof.Start),
-				End:   int(row.Proof.End),
-			}
+			tmpProof := nmt.NewInclusionProof(
```

Comment:
no need for `tmpProof` var, just assign to `proof`

Reply:
NewInclusion returns a copy.